### PR TITLE
cherrypick-1.1: sql: fix PARTITION BY multiple columns with window functions

### PR DESCRIPTION
**Pull request by nvb**
See #20151.

### CODE REVIEW

File: pkg/sql/logictest/testdata/logic_test/show_trace [hunk 1]
```diff
@@ -13,10 +13,10 @@ query TTT
 SELECT span, message, operation FROM [SHOW TRACE FOR SESSION] WHERE message NOT LIKE e'%\n%'
 ----
 (0,0)  === SPAN START: sql txn implicit ===  sql txn implicit
-(1,0)  === SPAN START: sql txn ===           sql txn implicit
-(1,0)  executing 1/5: BEGIN TRANSACTION      sql txn implicit
-(1,0)  executing 2/5: SELECT 1               sql txn implicit
-(1,0)  executing 3/5: COMMIT TRANSACTION     sql txn implicit
+(1,0)  === SPAN START: sql txn ===           sql txn
```
Comment (knz) [inline pkg/sql/logictest/testdata/logic_test/show_trace]: Wait why does this need to change? Seems unrelated.

Comment (nvanbenschoten) [reply pkg/sql/logictest/testdata/logic_test/show_trace]: Answered on the initial PR.

**Comment by knz** [comment]
LGTM.

cc @cockroachdb/release
